tests/testthat/test-nm-join.R

```diff
@@ -0,0 +1,156 @@
+skip_if_not_drone_or_metworx("test-nm-join")
+
+withr::local_options(list(
+  bbr.bbi_exe_path = read_bbi_path(),
+  bbr.verbose = FALSE
+))
+
+test_that("nm_join() works correctly with defaults and model object [BBR-NMJ-001]", {
+  withr::with_options(list(bbr.verbose = TRUE), {
+    expect_message({
+      test_df <- nm_join(MOD1)
+    }, regexp = "join stats")
+    expect_equal(nrow(test_df), DATA_TEST_ROWS_IGNORE)
+    expect_equal(ncol(test_df), DATA_TEST_COLS + TAB_NEW_COLS + PARTAB_NEW_COLS + 1)
+  })
+})
+
+test_that("nm_join() works correctly summary object [BBR-NMJ-001]", {
```

Comment:
This is minor and probably not worth worrying about, but I stumble when parsing the tail part of descriptions like this (there are a number in this PR).  I'm pretty sure you're wording it like that for the benefit of mrgvalprep (so that it can grab the common prefix), but I think ideally it could be done in a way that would still flow for readers of the test file.  Perhaps it could be something like "nm_join() works correctly: summary object", and mrgvalprep could learn to strip a trailing ":".